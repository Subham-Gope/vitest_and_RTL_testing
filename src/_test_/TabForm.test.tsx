// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import TabForm from "../components/TabForm";

// describe("Testing the TabForm.tsx file", () => {
//   test("let's user move focus from one input to next", async () => {
//     const { container } = render(
//       <TabForm message={`Hello there, this is a TabForm Component Test`} />
//     );

//     expect(container).toMatchSnapshot();

//     const input1 = screen.getByPlaceholderText("First name");
//     const input2 = screen.getByPlaceholderText("Last name");

//     input1.focus();
//     expect(input1).toHaveFocus();

//     await userEvent.tab();

//     expect(input2).toHaveFocus();
//   });
// });
