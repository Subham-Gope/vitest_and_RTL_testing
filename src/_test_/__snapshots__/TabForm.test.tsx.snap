// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the TabForm.tsx file > let's user move focus from one input to next 1`] = `
<div>
  <div>
    <p>
      Hello there, this is a TabForm Component Test
    </p>
    <div>
      <label
        for="firstName"
      >
        First Name
         
        <input
          id="firstName"
          placeholder="First name"
          type="text"
        />
      </label>
      <label
        for="lastName"
      >
        Last Name
         
        <input
          id="lastName"
          placeholder="Last name"
          type="text"
        />
      </label>
    </div>
  </div>
</div>
`;
